PukingPredator/Assets/Scripts/Physics/CollapseIfHeavyPlayer.cs

```diff
@@ -0,0 +1,34 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class CollapseIfHeavyPlayer : MonoBehaviour
+{
+    // Put on tiles with PhysicsBehavior and adjust the massThreshold
+
+    private PhysicsBehaviour pb;
+
+    /// <summary>
+    /// If player is greater than or equal to this mass when they collide with the structure, collapse the structure.
+    /// </summary>
+    [SerializeField]
+    private float massThreshold = 5f;
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        pb = GetComponent<PhysicsBehaviour>();
+    }
+
+    private void OnCollisionEnter(Collision collision)
+    {
+        if (collision.collider.CompareTag("Player"))
+        {
+            Inventory inventory = collision.collider.GetComponentInChildren<Inventory>();
+            if (inventory != null && inventory.getMass() >= massThreshold)
```

Comment:
if this is actually based on mass, it should just be using the players mass directly. if its based on item count, its fine to keep it as inventory, but access the property for it directly

Reply:
k I moved it to Player.cs, it's based on item count for now, but we can change it in the future